lib/train/transports/local.rb

```diff
@@ -36,6 +48,95 @@ def local?
         true
       end
 
+      def run_powershell_using_named_pipe(pipe_location, script)
+        pipe = nil
+        # Try to acquire pipe for 10 seconds with 0.1 second intervals.
+        # Removing this can result in instability due to the pipe being
+        # temporarily unavailable.
+        100.times do
+          begin
+            pipe = open(pipe_location, 'r+')
+            break
+          rescue
+            sleep 0.1
+          end
+        end
+        fail "Could not open pipe `#{pipe_location}`" if pipe.nil?
+        # Prevent progress stream from leaking to stderr
+        script = "$ProgressPreference='SilentlyContinue';" + script
+        encoded_script = Base64.strict_encode64(script)
+        pipe.puts(encoded_script)
+        pipe.flush
+        result = JSON.parse(Base64.decode64(pipe.readline))
+        pipe.close
+        result
+      end
+
+      def start_named_pipe_server(pipe_name) # rubocop:disable Metrics/MethodLength
+        require 'win32/process'
```

Comment:
Can we implement a fallback for windows platforms where this is not available, eg. windows nano?, in this case we have powershell available already.

Reply:
Is there a way I can identify Nano with current Train? If not I can fallback to not creating a pipe if I can't open one. Thoughts?